src/flightRecorder.cpp

```diff
@@ -1075,6 +1075,7 @@ class Recording {
         buf->putVar32(tid);
         buf->putVar32(call_trace_id);
         buf->putVar32(event->_thread_state);
+        buf->putVar64(Profiler::instance()->getContextId());
```

Comment:
> I apologise for complicating the matters

Nothing to apologies for, every comment that makes opensource software better is a good one. 

The other option would be to enable whole ```contextId``` feature by a flag that is disabled by default. If someone use ```contextId``` then there is a high chance that he/she is going to use different JFR viewer since JMC doesn't support ```contextId```. 

@apangin any thoughts on that?

Reply:
> If someone use contextId then there is a high chance that he/she is going to use different JFR viewer since JMC doesn't support contextId

`contextId` is written as a standard event attribute. Since JFR format is self-describing you will be able to see the `contextId` attribute value for events in JMC. 

I would really strongly suggest using custom event IDs (above 10000 eg. so we have a plenty of headroom for any builtin native events as well as any user defined Java events) - it would make it less of a hack. Let's face it - the events are going to have a different structure now so it is just fair that they are distinguishable from the builtin ones.

And, as I already mentioned - we still can keep all the metadata so eg. the execution samples are picked by JMC to present the profiling view etc.

Reply:
Looking forward to this feature will be released!

Regarding this discussion, I realized that converter's JfrReader would not work if context id is introduced, because though the JFR format itself is self-describing, JfrReader hard codes the parsing logic
https://github.com/async-profiler/async-profiler/blob/v2.9/src/converter/one/jfr/JfrReader.java#L155-L158

Should we fix JfrReader as well?